maubot/server.py

```diff
@@ -129,6 +131,22 @@ def setup_static_root_files(self, directory: str, ui_base: str) -> None:
             self.app.router.add_get(f"{ui_base}/{file}", lambda _: web.Response(body=data,
                                                                                 content_type=mime))
 
+        # also set up a resource path for the public url path prefix config
+        # cut the prefix path from public_url
+        public_url = self.config["server.public_url"]
+        base_path = self.config["server.base_path"]
+        public_url_path = ""
+        if public_url != "":
+            url_parts = urlparse(public_url)
+            public_url_path = url_parts.path.rstrip("/")
+
+        # assemble with base_path
+        api_path = f"{public_url_path}{base_path}"
+
+        path_prefix_response_body = json.dumps({"api_path": api_path.rstrip("/")})
```

Comment:
after changing to yarl:

```suggestion
        path_prefix_response_body = json.dumps({"api_path": str(api_path)})
```

Reply:
Would you prefer that I don't preemptively strip the trailing slash?

Reply:
Either way is fine, the `str()` was for converting the `yarl.URL` into a string.

Reply:
I think I prefer having this sort of cleanup done by the code instead of relying on user behaviour -- I get it's not that hard to get right either way, but as there are multiple moving parts (various paths and urls), I think overall we would benefit for the code to ensure the parts assemble nicely even in case of a forgotten slash somewhere.

If you don't have objections, I think I'd keep it that way.